torch/_inductor/sizevars.py

```diff
@@ -99,6 +99,8 @@ def simplify_with_ranges(expr: Expr, var_ranges: VarRanges) -> Expr:
             if result is None:
                 result = self._simplify_with_ranges(expr, var_ranges)
                 cache[key] = result
+                if result != expr:
+                    cache[(result, *var_ranges.items())] = result
```

Comment:
Do we see this happens that a simplified expression get passed in to sizevars.simplify_with_ranges again?

Reply:
Yeah, we simplify at lowering time then again at codegen time.